components/channel_notifications_modal/channel_notifications_modal.jsx

```diff
@@ -224,6 +237,16 @@ export default class ChannelNotificationsModal extends React.PureComponent {
         if (serverError) {
             serverErrorTag = <div className='form-group has-error'><label className='control-label'>{serverError}</label></div>;
         }
+        const desktopNotification = {
```

Comment:
This passes a new object into `NotificationSection` every time the modal re-renders which causes that to also re-render, even when it doesn't need to. I'd suggest we pass them in as separate props instead, but I think it might be worth not using `NotificationSection` for this at all. See below for what I mean.